pkgs/development/compilers/c3c/default.nix

```diff
@@ -3,24 +3,33 @@
 , fetchFromGitHub
 , cmake
 , python3
+, curl
+, libxml2
+, libffi
+, xar
 }:
 
 llvmPackages.stdenv.mkDerivation rec {
   pname = "c3c";
-  version = "unstable-2021-07-30";
+  version = "0.5";
 
   src = fetchFromGitHub {
     owner = "c3lang";
     repo = pname;
-    rev = "2246b641b16e581aec9059c8358858e10a548d94";
-    sha256 = "VdMKdQsedDQCnsmTxO4HnBj5GH/EThspnotvrAscSqE=";
+    rev = "3255183ee49cb903be1c66717d90f35cbcae7f82";
+    sha256 = "1482l0c8crirzaxqq9aghbvfj7c1bbbvzcyspmbv8xbxhyl48zcm";
   };
 
   nativeBuildInputs = [ cmake ];
 
   buildInputs = [
     llvmPackages.llvm
     llvmPackages.lld
+    curl
+    libxml2
+    libffi
+  ] ++ lib.optionals llvmPackages.stdenv.isDarwin [
+    xar
```

Comment:
Just to confirm, are these being added because of binaries they provide, or are they libraries to link against?

Reply:
What a great question and the answer took me on quite a ride and in turn made me modify allot of this PR.

Before this comment, I had the build statically linked (i.e. in a single c3c binary, so it didn't make sense to have the libs in buildInputs). Now I made it so it is dynamically linked, which in turn makes it so that we don't have to pass anymore flags to cmake and it also now makes sense to have these in buildInputs.